pkg/controllers/dynakube/extension/databases/deployment.go

```diff
@@ -0,0 +1,254 @@
+
+	userDataVolumeName = "user-data"
+	userDataMountPath  = "/var/userdata"
+	tokenVolumeName    = "auth-token"
+	tokenMountPath     = "/var/run/dynatrace/executor/token"
+	certsVolumeName    = "https-certs"
+	certsMountPath     = "/certs"
+)
+
+func buildAllLabels(dk *dynakube.DynaKube, dbex extensions.DatabaseSpec) (map[string]string, map[string]string, map[string]string) {
+	appLabels := labels.NewAppLabels(labels.DatabaseDatasourceLabel, dk.Name, labels.DatabaseDatasourceLabel, "")
+
+	deploymentLabels := appLabels.BuildLabels()
+	// Remove empty version label
+	delete(deploymentLabels, labels.AppVersionLabel)
+
+	matchLabels := appLabels.BuildMatchLabels()
+
+	templateLabels := maps.Clone(deploymentLabels)
+	templateLabels[executorLabelKey] = dbex.ID
+	templateLabels[datasourceLabelKey] = datasourceLabelValue
+
+	if dbex.Labels != nil {
+		temp := maps.Clone(dbex.Labels)
+		maps.Copy(temp, templateLabels)
+		templateLabels = temp
+	}
+
+	return deploymentLabels, matchLabels, templateLabels
+}
+
+func buildServiceAccountName(dbex extensions.DatabaseSpec) string {
+	if dbex.ServiceAccountName != "" {
+		return dbex.ServiceAccountName
+	}
+
+	return defaultServiceAccount
+}
+
+func buildContainer(dk *dynakube.DynaKube, dbex extensions.DatabaseSpec) corev1.Container {
+	pullPolicy := corev1.PullIfNotPresent
+	if dk.Spec.Templates.DatabaseExecutor.ImageRef.Tag == "latest" {
+		// For initial testing latest image is used, so let runtime pull updates if they're available.
+		// Maybe move this into the imageRef, e.g. imageRef.PullPolicy()
+		pullPolicy = corev1.PullAlways
+	}
+
+	container := corev1.Container{
+		Name:            "database-datasource",
+		Image:           dk.Spec.Templates.DatabaseExecutor.ImageRef.String(),
+		ImagePullPolicy: pullPolicy,
+		Args:            buildContainerArgs(dk),
+		Env:             buildContainerEnvs(),
+		LivenessProbe: &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path: "/health/live",
+					Port: intstr.IntOrString{IntVal: probePort},
+				},
+			},
+			InitialDelaySeconds: 15,
+			PeriodSeconds:       10,
+			TimeoutSeconds:      2,
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+		},
+		ReadinessProbe: &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path: "/health/ready",
+					Port: intstr.IntOrString{IntVal: probePort},
+				},
+			},
+			InitialDelaySeconds: 10,
+			PeriodSeconds:       5,
+			TimeoutSeconds:      2,
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+		},
+		Resources:       buildContainerResources(dbex.Resources),
+		SecurityContext: buildContainerSecurityContext(),
+		VolumeMounts:    buildVolumeMounts(dbex),
+	}
+
+	return container
+}
+
+func buildContainerArgs(dk *dynakube.DynaKube) []string {
+	return []string{
+		"--podid=$(POD_NAME)",
+		fmt.Sprintf("--url=https://%s:%d", dk.Extensions().GetServiceNameFQDN(), consts.OtelCollectorComPort),
```

Comment:
```suggestion
		fmt.Sprintf("--url=https://%s:%d", dk.Extensions().GetServiceNameFQDN(), consts.EecDatasourceComPort),
```
I thinks it's time to give this a more generic name. Not quite happy with my suggestion, though. It is about the port that datasources use to communicate with EEC.